### PR TITLE
8273998: Clarify specification for Window properties controlled by the window manager

### PR DESCRIPTION
**Pull request by kevinrushforth**
Update the API specification for the `Window` and `Stage` classes to clarify that the values for some properties and methods can be changed or ignored by the platform. Several of the properties already have a comment to the effect that the value can change externally, and thus the properties are not bindable, but this should be clarified further and for all such properties and methods.

See the Description of the JBS issues for this bug and its associated CSR for more details.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8273998](https://bugs.openjdk.java.net/browse/JDK-8273998): Clarify specification for Window properties controlled by the window manager
 * [JDK-8279620](https://bugs.openjdk.java.net/browse/JDK-8279620): Clarify specification for Window properties controlled by the window manager (**CSR**)

### Reviewers
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/708/head:pull/708` \
`$ git checkout pull/708`

Update a local copy of the PR: \
`$ git checkout pull/708` \
`$ git pull https://git.openjdk.java.net/jfx pull/708/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 708`

View PR using the GUI difftool: \
`$ git pr show -t 708`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/708.diff">https://git.openjdk.java.net/jfx/pull/708.diff</a>

</details>

### CODE REVIEW

**Comment by johanvos** [comment]
This looks ok to me. The clarifications make sense. There is unfortunately no guarantee about (low-level) cross-platform behaviour in these cases, but at least the JavaDoc makes this clear.

File: modules/javafx.graphics/src/main/java/javafx/stage/Window.java [hunk 1]
```diff
@@ -657,10 +682,6 @@ void notifySizeChanged(double newWidth, double newHeight) {
 
     /**
      * Whether or not this {@code Window} has the keyboard or input focus.
-     * <p>
```
Comment (aghaisas) [inline modules/javafx.graphics/src/main/java/javafx/stage/Window.java:660]: Is this removal intentional?

**Review by aghaisas — COMMENTED** [review]
Changes look good to me except at one place where you have removed the description. I will it out separately.

File: modules/javafx.graphics/src/main/java/javafx/stage/Window.java (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/javafx/stage/Window.java:660]: Yes, this is intentional. This property has no setter, so the existing comment is in error.

**Review by aghaisas — APPROVED** [review]

**Review by johanvos — APPROVED** [review]

**Comment by kevinrushforth** [comment]
This is intended for JavaFX 18, so I am changing the branch to `jfx18`, which will require a re-review by the reviewers.

**Review by johanvos — APPROVED** [review]
Re-approved

**Review by aghaisas — APPROVED** [review]
